Orange/widgets/unsupervised/owdistances.py

```diff
@@ -151,22 +169,34 @@ def _fix_missing():
             return True
 
         self.clear_messages()
-        if data is None:
-            return None
-        for check in (_check_sparse,
-                      _fix_discrete, _fix_missing, _fix_nonbinary):
-            if not check():
-                return None
-        try:
-            if metric.supports_normalization and self.normalized_dist:
-                return metric(data, axis=1 - self.axis, impute=True,
-                              normalize=True)
-            else:
-                return metric(data, axis=1 - self.axis, impute=True)
-        except ValueError as e:
+        if data is not None:
+            for check in (_check_sparse, _fix_discrete,
+                          _fix_missing, _fix_nonbinary):
+                if not check():
+                    data = None
+                    break
```

Comment:
Nothing to change here, just a comment: `all` (and `any`) shortcircuits. You could use `if not all(check() for check in ...): data = None`.